clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java

```diff
@@ -116,8 +116,8 @@
     /** indicates the minimum required inter broker magic required to support the API */
     public final byte minRequiredInterBrokerMagic;
 
-    /** indicates whether the API is enabled and should be exposed in ApiVersions **/
-    public final boolean isEnabled;
+    /** indicates whether this is an API which is only exposed by the KIP-500 controller **/
+    public final boolean isControllerOnlyApi;
```

Comment:
Could we get a JIRA to track the work for refactoring these boolean flags into static collections of api keys? To me it is not easy to use constructor correctly when there are multiple of them.

Reply:
Filed this JIRA: https://issues.apache.org/jira/browse/KAFKA-12232.